honeybee_3dm/togeometry.py

```diff
@@ -1,34 +1,309 @@
-"""Functions to create Ladybug geometries from Rhino geometries."""
+"""Functions to create Ladybug objects from Rhino3dm objects."""
 
-from ladybug_geometry.geometry3d.pointvector import Vector3D, Point3D
-from ladybug_geometry.geometry3d.line import LineSegment3D
-from ladybug_geometry.geometry3d.mesh import Mesh3D
-from ladybug_geometry.geometry3d.face import Face3D
-from ladybug_geometry.geometry3d.polyface import Polyface3D
+# The Rhino3dm library provides the ability to access content of a Rhino3dm
+# file from outside of Rhino
+import warnings
+import rhino3dm
 
+# Importing Ladybug geometry dependencies
 import ladybug.color as lbc
+from ladybug_geometry.geometry3d.pointvector import Point3D, Vector3D
+from ladybug_geometry.geometry3d.face import Face3D
+from ladybug_geometry.geometry3d.line import LineSegment3D
+from ladybug_geometry.geometry3d.polyline import Polyline3D
+from ladybug_geometry.geometry3d.mesh import Mesh3D
 
 
 def to_point3d(point):
-    """Ladybug Point3D from Rhino Point3d."""
-    pass
+    """Create a Ladybug Point3D object from a rhino3dm point.
+
+    Args:
+        point: A rhino3dm point.
+
+    Returns:
+        A Ladybug Point3D object.
+    """
+    return Point3D(point.X, point.Y, point.Z)
+
+
+def to_vector3d(vector):
+    """Create a Ladybug Vector3D from a rhino3dm Vector3d.
+
+    Args:
+        vector: A rhino3dm vector3d.
+
+    Returns:
+         A Ladybug Vector3D object.
+    """
+    return Vector3D(vector.X, vector.Y, vector.Z)
+
+
+def remove_dup_vertices(vertices, tolerance):
+    """Remove vertices from an array of Point3Ds that are equal within the tolerance.
+
+    Args:
+        vertices: A list of Ladybug Point3D objects.
+
+    Returns:
+         A list of Ladybug Point3D objects with duplicate points removed.
+
+    """
+    return [pt for i, pt in enumerate(vertices)
+            if not pt.is_equivalent(vertices[i - 1], tolerance)]
+
+
+def mesh_to_face3d(mesh):
+    """Create a Ladybug Face3D object out of a rhino3dm Mesh.
+
+    Args:
+        mesh: A rhino3dm mesh geometry.
+
+    Returns:
+        A list of Ladybug Face3D objects.
+    """
+
+    faces = []
+    pts = [to_point3d(mesh.Vertices[i]) for i in range(len(mesh.Vertices))]
+
+    for j in range(len(mesh.Faces)):
+        face = mesh.Faces[j]
+        if len(face) == 4:
+            all_verts = (pts[face[0]], pts[face[1]],
+                         pts[face[2]], pts[face[3]])
+        else:
+            all_verts = (pts[face[0]], pts[face[1]], pts[face[2]])
+
+        faces.append(Face3D(all_verts))
+
+    return faces
```

Comment:
Should we make a PolyFace instead? I _feel_ I already asked this and you told me the reason. Is it because Honeybee model does not support PolyFace?

Reply:
https://www.ladybug.tools/ladybug-geometry/docs/ladybug_geometry.geometry3d.polyface.html
I believe we can write a function `solid_to_polyface3d` and replace the function [solid_to_face3d](https://github.com/devngc/honeybee-3dm/blob/master/honeybee_3dm/togeometry.py#L234). But this function is not being used as of now since we're only dealing with faces. It will be useful when we start working with Honeybee Rooms.